presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1350,6 +1362,29 @@ else if (criteria instanceof JoinOn) {
             return output;
         }
 
+        private Expression findORExpression(Expression expression)
```

Comment:
It's a bit awkward having a BFS here. For a cleaner approach, take a look at ExpressionTreeUtils, I am pretty sure you can replace this whole BFS with a one-liner.

Reply:
If this query were:
`SELECT * FROM t1 JOIN t2 ON t1.a = t2.a AND (t2.b = t1.a OR t2.b = null)`
This would throw a warning. But should it? 
Since there is a key condition t1.a=t2.a, this shouldn't be a cross join. 
So I think the existing implementation with the BFS has the same issue.